### PR TITLE
[dhcp6relay] Add dhcpv6 option check (#10486)

### PR DESCRIPTION
**Pull request by kellyyeh**
(cherry picked from commit cfdb8431df86105698eabe9e5e1ee112c06fd7db)
(cherry picked from commit 2ead3aaefc1c20ff987faab6133accdff43de122)

#### Why I did it
To prevent relaying client packets with malformed payload

#### How I did it
Check for invalid dhcpv6 options

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

**Comment by kellyyeh** [comment]
Holding off merging until https://github.com/Azure/sonic-buildimage/issues/10785 is resolved

**Review by yxieca — APPROVED** [review]

File: src/dhcp6relay/src/relay.cpp [hunk 1]
```diff
@@ -622,21 +630,43 @@ void callback(evutil_socket_t fd, short event, void *arg) {
...
     switch (msg->msg_type) {
         case DHCPv6_MESSAGE_TYPE_RELAY_FORW:
         {
             relay_relay_forw(config->local_sock, current_position, ntohs(udp_header->len) - sizeof(udphdr), ip_header, config);
             break;
         }
-        default:
+        case DHCPv6_MESSAGE_TYPE_SOLICIT:
+        case DHCPv6_MESSAGE_TYPE_REQUEST: 
+	    case DHCPv6_MESSAGE_TYPE_CONFIRM:
```
Comment (lguohan) [inline src/dhcp6relay/src/relay.cpp:643]: make sure you have uniform identation

**Pull request merged**